Franchise Add-order

gauravThakur-sgl opened the pull request:
Currently adding local Storage for data Persistance

gauravThakur-sgl committed:
Added Route for Franchise order Details

gauravThakur-sgl committed:
Created ui for all pages

gauravThakur-sgl committed:
Changed Accordion Component

gauravThakur-sgl committed:
Added side information after completion of each page

gauravThakur-sgl committed:
Integrated country and states

gauravThakur-sgl committed:
Integrated api from shipping Rates and Validate Invoice

gauravThakur-sgl committed:
Added side Accordions value for the country name

gauravThakur-sgl committed:
Added Pricing information on side Accordion

gauravThakur-sgl committed:
Added Final Price Information SideBar

gauravThakur-sgl committed:
Updated Price Information

ChinmaySingh-ShipGlobal commented on src/components/Date.tsx:
```diff
@@ -1,8 +1,9 @@
 import { Control, Controller, FieldErrors } from "react-hook-form";
 import z from "zod";
-import { orderDetailsSchema } from "../zod/ordersSchema";
+import { orderDetailsSchema } from "../zod/franchiseOrderSchema";
```
use @ imports instead of "./" in al files

ChinmaySingh-ShipGlobal commented on src/components/Date.tsx:
```diff
@@ -1,8 +1,9 @@
+import { useState, useRef, useEffect } from "react";
+import { CalendarIcon } from "lucide-react";
 
 type FormData = z.infer<typeof orderDetailsSchema>;
```
make a file interfaces.ts and declare all interfaces and types in it and export them. For example- 

export type FormData = z.infer<typeof orderDetailsSchema>;

gauravThakur-sgl replied:
Done

gauravThakur-sgl committed:
Fixed product detail input, calender z-index,added currency in totalPrice

gauravThakur-sgl committed:
Added Validation on the iteam details

gauravThakur-sgl committed:
Added Address check on checkbox, fixed api check on every step

gauravThakur-sgl committed:
Fixed Information rendering on step 4

gauravThakur-sgl committed:
Fixed validateInvoice and shipperRate implementation logic

gauravThakur-sgl committed:
Added check on selected option

gauravThakur-sgl committed:
Added weight and items zod validation

gauravThakur-sgl committed:
Fixed conflicted background

gauravThakur-sgl committed:
Added interface and added shippingDataCard

gauravThakur-sgl committed:
Added spacing to the iteam Details

gauravThakur-sgl commented:
Fixed Information render issue on step 4
Validated itemdetails data
added check on selected option instead of individual data 
Added interface issue
by default 0% will be igst

ChinmaySingh-ShipGlobal commented on src/components/ui/Input.tsx:
```diff
@@ -57,7 +57,7 @@ function Input({
   return (
     <div>
       <div className="flex flex-col justify-start gap-2 appearance-none">
```
Never modify shadcn component file. Reset this

ChinmaySingh-ShipGlobal commented on src/page/franchise-order/AddOrder.tsx:
```diff
@@ -0,0 +1,189 @@
+  const [currentStep, setCurrentStep] = useState(1);
+  const [formData, setFormData] = useState({
+    consignorDetail: {
+      pickupAddress: "",
```
describe initial values in interfaces and use them in file with export

ChinmaySingh-ShipGlobal commented on src/page/franchise-order/AddOrder.tsx:
```diff
@@ -0,0 +1,189 @@
+
+  const stepTiles = ["consignorDetail", "consigneeDetail", "shipmentInformation", "selectShippingPartner"];
+
+  const handleNext = (data: ConsignorData | FormData | OrderDetailsFormData | ShippingPartnerFormData) => {
```
describe this as a type in interfaces and then use it here

ChinmaySingh-ShipGlobal commented on src/page/franchise-order/AddOrder.tsx:
```diff
@@ -0,0 +1,189 @@
+      setCurrentStep(currentStep + 1);
+    }
+    setOpenIndex((prev) => (prev !== null ? prev + 1 : null));
+    const nextAccordion = accordionRef.current?.querySelectorAll<HTMLDivElement>(".accordion")[currentStep];
```
for accordion use shadcn component

ChinmaySingh-ShipGlobal commented on src/page/franchise-order/AddOrder.tsx:
```diff
@@ -0,0 +1,189 @@
+        </p>
+        <div className="flex justify-center gap-4 mt-5">
+          <div className="flex flex-grow flex-col items-start gap-1 lg:w-2/3 scrollable-container">
+            <Accordion
```
Never name a component with name already present.

ChinmaySingh-ShipGlobal commented on src/page/franchise-order/ConsignorDetail.tsx:
```diff
@@ -0,0 +1,219 @@
+  });
+
+  const userDetail = [
+    {
```
declare constants at bottom of the file

ChinmaySingh-ShipGlobal commented on src/page/franchise-order/ConsignorDetail.tsx:
```diff
@@ -0,0 +1,219 @@
+  );
+};
+
+interface ISelectProps {
```
declare interfaces in separate interfaces.ts file

ChinmaySingh-ShipGlobal commented on src/page/franchise-order/ShipmentInformation.tsx:
```diff
@@ -0,0 +1,277 @@
+              render={({ field }) => (
+                <Select
+                  title="Invoice Currency"
+                  options={[
```
declare it as a constant and use here

ChinmaySingh-ShipGlobal commented on src/page/franchise-order/components/Accordion.tsx:
```diff
@@ -0,0 +1,54 @@
+  activeState?: number;
+}
+
+export const Accordion = ({ title, children, stepNumber, onToggle, isOpen, activeState = 0 }: AccordionItemProps) => {
```
use shadcn accordion component

gauravThakur-sgl committed:
Added seperate initital FormData file

gauravThakur-sgl committed:
Added ConsignorSelect Component

gauravThakur-sgl committed:
Added currency Options constant as a seperate file

gauravThakur-sgl committed:
fixed spacing for consignor select and item details

gauravThakur-sgl committed:
Added loader during price selection

gauravThakur-sgl committed:
Added priceInfo border

gauravThakur-sgl committed:
moved consignee Detail from the component to the page

gauravThakur-sgl committed:
Added Local storage for the data persistance

gauravThakur-sgl committed:
Fixed issues

gauravThakur-sgl committed:
Changed config to import using @

gauravThakur-sgl committed:
Added response message when no order found

gauravThakur-sgl committed:
Added autoFocus search and search icon on country select

gauravThakur-sgl committed:
Added getRate and validate data to the seperate File

gauravThakur-sgl committed:
Added Contact Card

gauravThakur-sgl committed:
Added responsiveness to the table

gauravThakur-sgl committed:
Fixed table border issue

gauravThakur-sgl committed:
Added safeList colors to the helper text

gauravThakur-sgl committed:
bug fix

gauravThakur-sgl committed:
Fixed state and country place holder and error message, added cursor-pointer and chevron on the information

gauravThakur-sgl committed:
Fixed Side Information

gauravThakur-sgl committed:
Added fixed measutrements sections

gauravThakur-sgl committed:
Fixed ui bugs

gauravThakur-sgl committed:
fixed country and state bug

gauravThakur-sgl committed:
fixed accordion title color

gauravThakur-sgl committed:
Fixed country state bug

gauravThakur-sgl committed:
Fixed Accordion step height and width

gauravThakur-sgl committed:
Fixed spacing in stepNumber

gauravThakur-sgl committed:
Fixed button padding and consignor detail

gauravThakur-sgl committed:
Fixed Consignor details content

gauravThakur-sgl committed:
Removed unused className and given space to thead

gauravThakur-sgl committed:
Combined multiple useState into one

gauravThakur-sgl committed:
Added @import to all pages and components

gauravThakur-sgl committed:
Added @ import in the app and main file

gauravThakur-sgl committed:
Added missing imports

gauravThakur-sgl committed:
Added token

gauravThakur-sgl committed:
Removed check on shipping and billing details on submit

gauravThakur-sgl committed:
updated state and country error message

gauravThakur-sgl committed:
Updated imports
